scrappers/itaka/scraper.py

```diff
@@ -0,0 +1,202 @@
+        return final_rating
+
+    def _get_img(self, html_content: BeautifulSoup) -> str:
+        gallery_content = str(html_content.find("div", {"id": "gallery"}))
+
+        string_starting_point = gallery_content.find("<meta content=")
+        string_end_point = gallery_content.find(
+            "'", string_starting_point + len("<meta content='")
+        )
+
+        link_start_position = string_starting_point + len("<meta content='")
+        link_end_position = string_end_point
+        return gallery_content[link_start_position:link_end_position].split('"', 1)[0]
+
+    def _room_picker(self, html_content: BeautifulSoup) -> Tuple[bool, ...]:
+        input_string = (
+            html_content.find("div", {"id": "product-tab-productdescription"})
+            .text.strip()
+            .split("SPORT I ROZRYWKA:")[0]
+            .strip()
+            .split("POKÓJ:")[-1]
+            .strip()
+        )
+
+        room_options = {
+            "standardowy": False,
+            "rodzinny": False,
+            "apartament": False,
+            "suite": False,
+        }
+        room_options = {option: option in input_string for option in room_options}
+        return tuple(room_options.values())
+
+    def generate_data(self) -> List[dict]:
+        iterations = 0
+        scraped_data = []
+        for offer_url in self.offer_urls:
+            html_page_source = requests.get(offer_url)
+            html_content = BeautifulSoup(html_page_source.content, "html.parser")
+
+            hotel_name = self._get_hotel_name(html_content)
+            country_name = self._get_country_name(html_content)
+            city_name = self._get_city_name(html_content)
+            description = self._get_description(html_content)
+            score = self._get_score(html_content)
+            link = self._get_img(html_content)
+            (
+                is_standard,
+                is_family,
+                is_apartament,
+                is_studio,
+            ) = self._room_picker(html_content)
+
+            json_entry = {
+                "operator": "Itaka",
+                "hotel": hotel_name,
+                "country": country_name,
+                "city": city_name,
+                "description": description,
+                "score": score,
+                "img": link,
+                "room": {
+                    "is_standard": is_standard,
+                    "is_family": is_family,
+                    "is_apartment": is_apartament,
+                    "is_studio": is_studio,
+                },
+            }
+
+            scraped_data.append(json_entry)
+            print(f"--------- TRIP #{iterations + 1}---------")
+            iterations += 1
+
+        return scraped_data
+
+    def save_json_as_file(self, filename: str) -> None:
+        with open(f"{filename}.json", "w") as outfile:
+            json.dump(self.scraped_data, outfile)
+
+
+scraper = ItakaScraper(WEBSITE_URL, NO_OF_SCRAPED_OFFERS)
```

Comment:
Dobrym nawykiem przy skryptach jest stosowanie takiej konwencji
```suggestion
def main(*comand_line_args: str) -> None:
    scraper = ItakaScraper(WEBSITE_URL, NO_OF_SCRAPED_OFFERS)
    scrapper.save_json_as_file("itaka_offers")
       
if __name__ == "__main__":
    main()
```